### PR TITLE
#86: Wiring Up

### LINKED ISSUE #86: Change Requests - Wire Up N Proposed Solutions
## Description

Wire up the new n proposed solutions form to use relevant endpoints, where necessary.

## Desired Behavior

From the list of proposed solutions, when you create a modal to add in a new proposed solution, send in details of the modal to the backend database.
These details are;
- [ ] The description for the proposed solution, which should be a string
- [ ] The Timeline impact, in number of weeks, which should be an integer
- [ ] The budget impact, in dollars, which should be an integer
- [ ] The scope of the change, which should be a string

### PR DESCRIPTION
**Pull request by anthonybernardi**
## Changes

wired up everything! This is the final proposed solution experience tm as it's the last thing we need to finish!

closes #86

### CODE REVIEW

**Review by Ari-S-123 — APPROVED** [review]
lgtm

File: src/frontend/src/pages/ChangeRequestDetailPage/ProposedSolutionsList.tsx [hunk 1]
```diff
@@ -51,7 +51,7 @@ const ProposedSolutionsList: React.FC<ProposedSolutionsListProps> = ({
 
   return (
     <>
-      {!crReviewed ? (
+      {crReviewed === undefined ? (
```
Comment (RChandler234) [inline src/frontend/src/pages/ChangeRequestDetailPage/ProposedSolutionsList.tsx:54]: What's going on here

File: src/frontend/src/pages/ChangeRequestDetailPage/ProposedSolutionForm.tsx [hunk 2]
```diff
@@ -63,7 +63,14 @@ const ProposedSolutionForm: React.FC<ProposedSolutionFormProps> = ({
 
   return (
     <PageBlock title="" cardContainerStyle="mb-0">
-      <Form id="individual-proposed-solution-form" onSubmit={handleSubmit(onAdd)}>
+      <Form
+        id="individual-proposed-solution-form"
+        onSubmit={(e) => {
+          e.preventDefault();
+          e.stopPropagation();
```
[the remark this answers is not in the document]
Comment (anthonybernardi) [reply src/frontend/src/pages/ChangeRequestDetailPage/ProposedSolutionForm.tsx:70]: my understanding is that e.preventdefault stops the default behavior for this inner form and stoppropagation stops the outer form from being submitted to afterwards as well

File: src/frontend/src/pages/ChangeRequestDetailPage/ProposedSolutionsList.tsx (hunk 1 above)
Comment (anthonybernardi) [reply src/frontend/src/pages/ChangeRequestDetailPage/ProposedSolutionsList.tsx:54]: if crreviewed is undefined/null, that means the change request hasn't been reviewed at all. if it's true, it's approved, if it's false, it's denied. we can only see the add prposed solution button when a CR hasn't been reviewed yet

File: src/frontend/src/pages/ChangeRequestDetailPage/ProposedSolutionForm.tsx (hunk 2 above)
[the remark this answers is not in the document]
Comment (anthonybernardi) [reply src/frontend/src/pages/ChangeRequestDetailPage/ProposedSolutionForm.tsx:70]: just tested without to be sure and yeah it's needed

File: src/frontend/src/pages/CreateChangeRequestPage/CreateProposedSolutionsList.tsx [hunk 3]
```diff
@@ -10,19 +10,24 @@ import { Button, Modal } from 'react-bootstrap';
...
+
+const CreateProposedSolutionsList: React.FC<CreateProposedSolutionsListProps> = ({
+  proposedSolutions,
+  setProposedSolutions
+}) => {
   const [showEditableForm, setShowEditableForm] = useState<boolean>(false);
 
   const addProposedSolution = async (data: ProposedSolution) => {
-    proposedSolutionsList.push(data);
+    proposedSolutions.push(data);
```
[the remark this answers is not in the document]
Comment (anthonybernardi) [reply src/frontend/src/pages/CreateChangeRequestPage/CreateProposedSolutionsList.tsx:25]: true

**Review by callmejim1226 — COMMENTED** [review]
Left a comment in the ReviewChangeRequestView file! The rest changes looked all good to me. Good job!

**Pull request merged**